src/Storage/custom/hybrid-2019-03-01/New-AzStorageAccount.ps1

```diff
@@ -0,0 +1,338 @@
+        [Parameter(ParameterSetName='CreateExpanded1KeyVaultEncryption', HelpMessage='Required for storage accounts where kind = BlobStorage. The access tier used for billing.')]
+        [Parameter(ParameterSetName='CreateViaIdentityExpanded1StorageEncryption', HelpMessage='Required for storage accounts where kind = BlobStorage. The access tier used for billing.')]
+        [Parameter(ParameterSetName='CreateViaIdentityExpanded1KeyVaultEncryption', HelpMessage='Required for storage accounts where kind = BlobStorage. The access tier used for billing.')]
+        [ArgumentCompleter([Microsoft.Azure.PowerShell.Cmdlets.Storage.Support.AccessTier])]
+        [Microsoft.Azure.PowerShell.Cmdlets.Storage.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Storage.Support.AccessTier]
+        # Required for storage accounts where kind = BlobStorage. The access tier used for billing.
+        ${AccessTier},
+    
+        [Parameter(ParameterSetName='CreateExpanded1StorageEncryption', HelpMessage='A boolean indicating whether or not the service encrypts the data as it is stored.')]
+        [Parameter(ParameterSetName='CreateExpanded1KeyVaultEncryption', HelpMessage='A boolean indicating whether or not the service encrypts the data as it is stored.')]
+        [Parameter(ParameterSetName='CreateViaIdentityExpanded1StorageEncryption', HelpMessage='A boolean indicating whether or not the service encrypts the data as it is stored.')]
+        [Parameter(ParameterSetName='CreateViaIdentityExpanded1KeyVaultEncryption', HelpMessage='A boolean indicating whether or not the service encrypts the data as it is stored.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Storage.Category('Body')]
+        [System.Management.Automation.SwitchParameter]
+        # A boolean indicating whether or not the service encrypts the data as it is stored.
+        ${BlobEnabled},
+    
+        [Parameter(ParameterSetName='CreateExpanded1StorageEncryption', Mandatory, HelpMessage='Gets or sets the custom domain name assigned to the storage account. Name is the CNAME source.')]
+        [Parameter(ParameterSetName='CreateExpanded1KeyVaultEncryption', Mandatory, HelpMessage='Gets or sets the custom domain name assigned to the storage account. Name is the CNAME source.')]
+        [Parameter(ParameterSetName='CreateViaIdentityExpanded1StorageEncryption', Mandatory, HelpMessage='Gets or sets the custom domain name assigned to the storage account. Name is the CNAME source.')]
+        [Parameter(ParameterSetName='CreateViaIdentityExpanded1KeyVaultEncryption', Mandatory, HelpMessage='Gets or sets the custom domain name assigned to the storage account. Name is the CNAME source.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Storage.Category('Body')]
+        [System.String]
+        # Gets or sets the custom domain name assigned to the storage account. Name is the CNAME source.
+        ${CustomDomainName},
```

Comment:
It should be because the Name property is mandatory but CustomDomain object is not. I filled an issue: https://github.com/Azure/autorest.powershell/issues/408

Reply:
But, since this is a customization, we can make it optional, can't we?